cellrank/tools/_utils.py

```diff
@@ -1586,41 +1596,43 @@ def _solve_many_sparse_problems(
     solver: LinSolver,
     tol: float = 1e-5,
     related_columns_in_b: Optional[Iterable] = None,
-):
+) -> Tuple[np.ndarray, List[int], int]:
     """
-    Solve `mat_a*x = mat_b` efficiently using an iterative solver.
+    Solve `mat_a * x = mat_b` efficiently using an iterative solver.
 
     This is a utility function which is optimized for the case of `mat_a` and `mat_b` being sparse,
     and columns in `mat_b` being related. In that case, we can treat each column of `mat_b` as a
-    seperate linear problem and solve that efficiently using iterative solvers that exploit sparsity.
-    If the columsn of `mat_b` are related, we can use the solution of the previous problem as an
+    separate linear problem and solve that efficiently using iterative solvers that exploit sparsity.
+
+    If the columns of `mat_b` are related, we can use the solution of the previous problem as an
     initial guess for the next problem. Further, we parallelize the individual problems for each
     column in `mat_b` and solve them on separate kernels.
 
-
-    Parameters
-    --------
+    Params
+    ------
     mat_a
         Matrix of shape `n x n`. We make no assumptions on `mat_a` being symmetric or positive definite.
     mat_b
         Matrix of shape `n x m`, with m << n.
     solver
-        Solver to use for the linear problem. Options are [gmres, lgmres, bicgstab, gcrotmk]. These
-        can be found in `scipy.sparse.linalg`.
+        Solver to use for the linear problem. Options are `['gmres', 'lgmres', 'bicgstab', 'gcrotmk']`.
+        These can be found in :func:`scipy.sparse.linalg`.
```

Comment:
Could also benchmark these different solvers, but I wouldn't spend much time on it, maybe just on a single problem to see whether there is a clear winner.